modules/ROOT/pages/dataweave-streaming.adoc

```diff
@@ -0,0 +1,398 @@
+* `input` directive: +
+The `@StreamCapable()` annotation requires the use of an input directive in the DataWeave script that identifies the MIME type of the data source, for example, `input payload application/xml`.
+
+The validator checks a script against the following criteria:
+
+* The variable is referenced only once
+* No index selector is set for negative access, such as `[-1]`
+* No reference to the variable is found in a nested lambda
+
+If all are true, the selected data is streamable. The following example validates successfully. The script is designed to act on the <<example_json, JSON input>> from the <<stream_json, JSON streaming>> section.
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+
+@StreamCapable()
+input payload application/json streaming=true
+output application/json
+---
+payload.family filter (member) -> member.age > 3
+----
+
+The script successfully validates and returns the following output:
+
+[source,JSON,linenums]
+----
+[
+  {
+    "name": "Pedro",
+    "age": 4
+  },
+  {
+    "name": "Matias",
+    "age": 8
+  }
+]
+----
+
+=== Validation Failures
+
+If any of the criteria that the valiator checks are false, the validation fails. 
+
+Before proceeding, note that validation can fail in some cases when streaming works. If you write a script in a way that sequentially accesses the input variable in a given data source, streaming works, but that script might not work in all cases. For example, JSON does not place a restriction on the order of the keys in an object. If the keys in some JSON documents arrive in a different order than the script expects, streaming will fail in those cases. The annotation processor follows the rules of the format and cannot assume that the keys always arrive in the same order.
+
+==== Error: Variable is Referenced More Than Once
+
+Validation fails if a script attempts to reference the same variable more than once.
+
+The following script is designed to act on the <<example_json, JSON input>> from the <<stream_json, JSON streaming>> section. Validation fails because the script attempts to reference the `payload` variable more than once.
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+
+@StreamCapable()
+input payload application/json
+output application/json
+---
+ {
+     family: payload.family filter (member) -> member.age > 3,
+     name: payload.name
+ }
+----
+
+The script fails with the following error:
+
+[source,error,linenums]
+----
+4| input payload application/json streaming=true
+         ^^^^^^^
+Parameter `payload` is not stream capable.
+Reasons:
+ - Variable payload is referenced more than once. Locations:
+---------------------------
+
+8|      family: payload.family filter (member) -> member.age > 3,
+                ^^^^^^^
+---------------------------
+
+9|      name: payload.name
+              ^^^^^^^ at
+4| input payload application/json streaming=true
+----
+
+
+==== Error: Wrong Scope Reference
+
```

Comment:
ditto: why document the user's failure to heed prerequisites, notices, and other clearly documented requirements?

Reply:
We're showing the error that is produced when you use the annotation but fail to meet the requirements of the annotation processor.  If they get that error, they'll understand why.